cleanup_utils/mysql_cleanup.py

```diff
@@ -0,0 +1,136 @@
+            user=database_url_obj.username or '',
+            passwd=database_url_obj.password or '',
+            port=database_url_obj.port or 3306,
+        )
+    except MySQLError as exc:
+        logger.exception('Cannot get MySQL cursor: %s', exc)
+        raise
+    return connection.cursor()
+
+
+# Classes ####################################################################
+
+class MySqlCleanupInstance:
+    """
+    Handles the cleanup of old MySQL databases
+    """
+    def __init__(self, age_limit, url, domain, dry_run):
+        """
+        Set up variables needed for cleanup
+        """
+        self.age_limit = age_limit
+        self.cleaned_up_hashes = []
+        self.domain_suffix = domain.replace('.', '_')
+        self.dry_run = dry_run
+        try:
+            self.cursor = _get_cursor(url)
+        except Exception:  # pylint: disable=broad-except
+            # Exception has already been logged
+            self.cursor = None
+
+    def _get_old_databases(self):
+        """
+        Returns a list of databases older than the age limit
+        """
+        query = """SELECT table_schema, MAX(create_time) AS create_time
+            FROM information_schema.tables
+            WHERE create_time <= DATE_SUB(NOW(), INTERVAL %(age_limit)s DAY)
+            AND table_schema LIKE %(domain_filter)s
+            GROUP BY table_schema ORDER BY create_time DESC"""
+        params = {
+            'age_limit': self.age_limit,
+            'domain_filter': '%_{}_%'.format(self.domain_suffix)
+        }
+        self.cursor.execute(query, params)
+        return self.cursor.fetchall()
+
+    def run_cleanup(self):
+        """
+        Runs the cleanup of MySQL databases older than the age limit
+        """
+        logger.info("\n --- Starting MySQL Cleanup ---")
+        if not self.cursor:
+            logger.error(
+                'ERROR: Not connected to the database, '
+                'MySQL cleanup will be skipped.'
+            )
+            return
+
+        if self.dry_run:
+            logger.info("Running in DRY_RUN mode, no actions will be taken.")
+
+        databases = self._get_old_databases()
+        logger.info('Found %d old databases', len(databases))
+
+        for (database, create_date) in databases:
+            logger.info('  > Considering database %s', database)
+            instance_database_re = r'^([0-9a-f]{8})([_0-9a-z]+)?_%s' % (self.domain_suffix,)
+            match = re.match(instance_database_re, database)
+            if not match:
+                logger.info(
+                    '    * SKIPPING: Did not match expected format %r.',
+                    instance_database_re
+                )
+                continue
+
+            logger.info(
+                '    * Dropping MySQL DB `%s` created at %s', database,
+                datetime.strftime(create_date, '%Y-%m-%dT%H:%M:%SZ'))
+            if not self.dry_run:
+                try:
+                    self.cursor.execute(
```

Comment:
@jamestait, I am wondering if we need to do this one by one instead of using `executemany()` to do it in one shot. I know that the performance of the `executemany()` method varies from driver to driver and also based on the backend, but can you check and find out if it makes sense to use it?

Reply:
From the docstring for `MySQLdb.cursors.executemany()`:

    """Execute a multi-row query.

    :param query: query to execute on server
    :param args:  Sequence of sequences or mappings.  It is used as parameter.
    :return: Number of rows affected, if any.

    This method improves performance on multiple-row INSERT and
    REPLACE. Otherwise it is equivalent to looping over args with
    execute().
    """

So I don't think we gain anything with `executemany()` but we'd lose the ability to report individually on each operation.